samples/WebsocketConnect/README.md

```diff
@@ -48,3 +48,139 @@ If you wish to use the latest SDK release to run the sample rather than using th
+            .withProtocolOperationTimeoutMs(60000);
+
+        builder.withWebsockets(true);
+        builder.withWebsocketSigningRegion(cmdData.input_signingRegion);
+
+        StaticCredentialsProviderBuilder providerBuilder = new StaticCredentialsProviderBuilder();
+        providerBuilder.withAccessKeyId("<access key id>");
+        providerBuilder.withSecretAccessKey("<secret access key>");
+        providerBuilder.withSessionToken("<session>");
+
+        CredentialsProvider credentialsProvider = providerBuilder.build();
+        builder.withWebsocketCredentialsProvider(credentialsProvider);
+
+        MqttClientConnection connection = builder.build();
+        return connection;
+    }
+}
+```
+
+### MQTT over WebSockets with Custom Authorizer
+
+An MQTT3 direct connection can be made using a [Custom Authorizer](https://docs.aws.amazon.com/iot/latest/developerguide/custom-authentication.html).
+When making a connection to a Custom Authorizer, the MQTT3 client can optionally passing username, password, and/or token
+signature arguments based on the configuration of the Custom Authorizer on AWS IoT Core.
+
+You will need to setup your Custom Authorizer so that the lambda function returns a policy document to properly connect.
+See [this page](https://docs.aws.amazon.com/iot/latest/developerguide/config-custom-auth.html) on the documentation for
+more details and example return results.
+
+If your Custom Authorizer does not use signing, you don't specify anything related to the token signature and can use
+the following code:
+
+```java
+static MqttClientConnection createMqttClientConnection(CommandLineUtils.SampleCommandLineData cmdData) {
+    try (AwsIotMqttConnectionBuilder builder = AwsIotMqttConnectionBuilder.newDefaultBuilder()) {
+        if (cmdData.input_ca != "") {
+            builder.withCertificateAuthorityFromPath(null, cmdData.input_ca);
+        }
+        builder.withConnectionEventCallbacks(callbacks)
+            .withClientId(cmdData.input_clientId)
+            .withEndpoint(cmdData.input_endpoint)
+            .withPort(cmdData.input_port)
+            .withCleanSession(true)
+            .withProtocolOperationTimeoutMs(60000);
+        builder.withCustomAuthorizer(
+            cmdData.input_customAuthUsername,
+            cmdData.input_customAuthorizerName,
+            null,
+            cmdData.input_customAuthPassword,
+            null,
+            null);
+        builder.withWebsockets(true);
+        builder.withWebsocketSigningRegion(cmdData.input_signingRegion);
+        MqttClientConnection connection = builder.build();
+        return connection;
+    } catch (Exception ex) {
+        throw new RuntimeException("Failed to create MQTT311 connection", ex);
+    }
+}
+```
+
+To run the websocket connect with custom authorizer use the following command:
+
+```sh
+mvn compile exec:java -pl samples/WebsocketConnect -Dexec.mainClass=websocketconnect.WebsocketConnect -Dexec.args="\
```

Comment:
I don't think these command line arguments are setup for the basic WebsocketConnect sample. e.g. `custom_auth_username`, `custom_auth_authorizer_name`, `custom_auth_password`. The websocket sample currently loads cmdData using "CommandLineUtils.SampleCommandLineData cmdData = CommandLineUtils.getInputForIoTSample("WebsocketConnect", args);" which in the util doesn't setup those specific arguments. We'll need to either build a specific separate sample that the CommandLineUtils loads and sets these arguments for, or we provide snippets for directly setting these arguments without use of cmdData.

Reply:
I removed `cmdData`, so "How to run" became redundant